External/FEXCore/Source/Interface/HLE/Thunks/Thunks.cpp

```diff
@@ -67,6 +69,107 @@ extern char __start_HostToGuestTrampolineTemplate[];
 extern char __stop_HostToGuestTrampolineTemplate[];
 
 namespace FEXCore {
+  namespace VDSO {
+    using TimeType = decltype(::time)*;
+    using GetTimeOfDayType = decltype(::gettimeofday)*;
+    using ClockGetTimeType = decltype(::clock_gettime)*;
+    using ClockGetResType = decltype(::clock_getres)*;
+    using GetCPUType = decltype(::getcpu)*;
+
+    TimeType TimePtr = ::time;
+    GetTimeOfDayType GetTimeOfDayPtr = ::gettimeofday;
+    ClockGetTimeType ClockGetTimePtr = ::clock_gettime;
+    ClockGetResType ClockGetResPtr = ::clock_getres;
+    GetCPUType GetCPUPtr = ::getcpu;
+
+    static void time(void* ArgsRV) {
+      struct ArgsRV_t {
+        time_t *a_0;
+        uint64_t rv;
+      } *args = reinterpret_cast<ArgsRV_t*>(ArgsRV);
+
+      args->rv = TimePtr(args->a_0);
+    }
+
+    static void gettimeofday(void* ArgsRV) {
+      struct ArgsRV_t {
+        struct timeval *tv;
+        struct timezone *tz;
+        uint64_t rv;
+      } *args = reinterpret_cast<ArgsRV_t*>(ArgsRV);
+
+      args->rv = GetTimeOfDayPtr(args->tv, args->tz);
+    }
+
+    static void clock_gettime(void* ArgsRV) {
+      struct ArgsRV_t {
+        clockid_t clk_id;
+        struct timespec *tp;
+        uint64_t rv;
+      } *args = reinterpret_cast<ArgsRV_t*>(ArgsRV);
+
+      args->rv = ClockGetTimePtr(args->clk_id, args->tp);
+    }
+
+    static void clock_getres(void* ArgsRV) {
+      struct ArgsRV_t {
+        clockid_t clk_id;
+        struct timespec *tp;
+        uint64_t rv;
+      } *args = reinterpret_cast<ArgsRV_t*>(ArgsRV);
+
+      args->rv = ClockGetResPtr(args->clk_id, args->tp);
+    }
+
+    static void getcpu(void* ArgsRV) {
+      struct ArgsRV_t {
+        uint32_t *cpu;
+        uint32_t *node;
+        uint64_t rv;
+      } *args = reinterpret_cast<ArgsRV_t*>(ArgsRV);
+
+      args->rv = GetCPUPtr(args->cpu, args->node);
+    }
+
+    void LoadVDSO() {
+      void *vdso = dlopen("linux-vdso.so.1", RTLD_LAZY | RTLD_LOCAL | RTLD_NOLOAD);
+      if (!vdso) {
+        vdso = dlopen("linux-gate.so.1", RTLD_LAZY | RTLD_LOCAL | RTLD_NOLOAD);
+      }
+
+      if (!vdso) {
+        // Couldn't load VDSO
```

Comment:
Adding an informational message, since even if we can't load the host VDSO (uncommon but can happen on some ARM configurations) then we still want to pass the guest down an optimal libc implementation path still.